In [1]:
from IPython.display import HTML

In [2]:
import json
import requests
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy


# Testing Sites


In [3]:
testing_sites_df =  pd.read_csv('testingsites.csv')

In [4]:
testing_sites_df.drop(columns = ['Owner Name','Phone','Website','Operational Hours','Vehicle Capacity','CreationDate','Drive-through','Appointment Only','Referral Required','Call first','Virtual/Telehealth Screening','Local Health Department URL','Vehicle Required','Offering same-day diagnostic (not antibody) results','Offering Take-Home Test','Status Color'], inplace = True)
testing_sites_df.drop(columns = ['Public Form Submission','Minimum Age','Wheelchair accessibility ','Accessibility Support'], inplace = True)

In [5]:

testing_sites_df['county']= testing_sites_df['County']
testing_sites_df
testing_sites_df.sort_values(by ='County')


Name of Facility  \
8858                                         CHC Enfield   
6809                                      ALBERTSONS 184   
4850                  Family Medicine Health Center-Kuna   
6802                                      ALBERTSONS 169   
3895              Rite Aid/Project Baseline Testing Site   
6798                                      ALBERTSONS 162   
6797                                      ALBERTSONS 164   
662                 Primary Health - Meridian Crossroads   
6795                                      ALBERTSONS 160   
6793                                      ALBERTSONS 154   
6792                                      ALBERTSONS 156   
6790                                     ALBERTSONS 3339   
6789                                     ALBERTSONS 3337   
6786                                      ALBERTSONS 193   
6783                                      ALBERTSONS 199   
6779                                      ALBERTSONS 101   
1296   Crush the Curve Antibody Testing at Ten Mile C...   
6722                             Ridley's Pharmacy #8448   
6718                        Ridely's Star Pharmacy #8440   
98                                      St. Luke's Plaza   
97                          St. Luke's Meridian Hospital   
94                 Saint Alphonsus Meridian Health Plaza   
4202                           St Luke's Clinic-Meridian   
660                         Primary Health - Garden City   
4849              Family Medicine Health Center-Meridian   
4848               Family Medicine Health Center-Raymond   
6796                                      ALBERTSONS 161   
4200                              St Luke's Clinic-Boise   
4847             Family Medicine Health Center - Emerald   
6812                                  MARKET STREET 3195   
...                                                  ...   
10667                                        Hy-Vee East   
10668                                       South Hy-Vee   
10669                           Ansborough Avenue Hy-Vee   
10670                                      Waukee Hy-Vee   
10671                                 Valley West Hy-Vee   
10672                 St. Charles Hy-Vee HealthMarket Rx   
10673                Stewartville Hy-Vee HealthMarket Rx   
10674                                   Maryville Hy-Vee   
10675                                    Columbus Hy-Vee   
10676                                Grand Island Hy-Vee   
10677                                     Kearney Hy-Vee   
10707                             Holy Redeemer Hospital   
10708                  Temple University - Ambler Campus   
10709                vybe urgent care - Center City East   
10710                       vybe urgent care - Havertown   
10711                       Jefferson Health - Navy Yard   
10712  American Family Care (AFC) Urgent Care - Chelt...   
10713       CHOP Brandywine Valley Specialty Care Center   
10714                    Tower Health (Plymouth Meeting)   
10715           Philadelphia FIGHT - Mifflin Square Park   
10783                                   Lynn Urgent Care   
10784                  Reliant Medical Group Marlborough   
10785                   Project Beacon - Stop the Spread   
10786  Greater Lawrence Family Health Center - S. Law...   
10787  Greater Lawrence Family Health Center - Arling...   
10788                              CIC Health Middlebury   
10789                             CIC Health Brattleboro   
10790                                  CIC Health Newton   
10791                               CIC Health Cambridge   
10792                              CIC Health Providence   

                                            Full Address      Municipality  \
8858                    5 North Main Street, Enfield, CT           Enfield   
6809             6560 SOUTH FEDERAL WAY, Boise ID, 83716             Boise   
4850   708 E Wythe Creek Ct Ste 103, Kuna, ID 83634-5005              Kuna   
6802   

In [6]:
testing_sites_df['county_states']= testing_sites_df['county'] + ' , ' + testing_sites_df['State or Territory']
testing_sites_df

Name of Facility  \
0      Wheeling Hospital/WVU Hospitals Offsite COVID-...   
1                              Middleton Family Medicine   
2                       Central Outreach Wellness Center   
3                          Anthony Wayne Recreation Area   
4      Wexford Health + Wellness Pavilion (Allegheny ...   
5                                      Abington Hospital   
6      CVS Health COVID-19 Drive Thru Testing Site - ...   
7                         Forest Community Health Center   
8                                        Redicare Okemos   
9         LVHN COVID-19 Assess and Test - MacArthur Road   
10                                  PNC Bank Arts Center   
11           LVHN COVID-19 Assess and Test - Stroudsburg   
12             LVHN COVID-19 Assess and Test - Palmerton   
13              LVHN COVID-19 Assess and Test - Macungie   
14     LVHN COVID-19 Assess and Test - Richland Township   
15                                Bassett Medical Center   
16                LVHN COVID-19 Assess and Test–Hazleton   
17                 VHN COVID-19 Assess and Test–Cressona   
18                          Metroparks Hawkins Farmhouse   
19                LVHN COVID-19 Assess and Test–Nazareth   
20             LVHN COVID-19 Assess and Test–17th Street   
21              LVHN COVID-19 Assess and Test–Muhlenberg   
22               Beacon MedPoint Urgent Care Main Street   
23            LVHN COVID-19 Assess and Test–Bartonsville   
24                            Saint Joseph Health System   
25                                 Champlain Valley Expo   
26                Vermont Department of Health Labratory   
27                Good Samaritan Regional Medical Center   
28                     Samaritan Albany General Hospital   
29                        Jefferson Univsersity Hospital   
...                                                  ...   
10774                                     S.A.N.D School   
10775                             Bloomfield High School   
10776                                    Xfinity Theatre   
10777                 Dr Michael Fox School Testing Site   
10778                                   Columbus Academy   
10779                  Gerald Ford International Airport   
10780                        Enumclaw Municipal Building   
10781                                 Riverside MBTA lot   
10782                           Veterans Memorial School   
10783                                   Lynn Urgent Care   
10784                  Reliant Medical Group Marlborough   
10785                   Project Beacon - Stop the Spread   
10786  Greater Lawrence Family Health Center - S. Law...   
10787  Greater Lawrence Family Health Center - Arling...   
10788                              CIC Health Middlebury   
10789                             CIC Health Brattleboro   
10790                                  CIC Health Newton   
10791                               CIC Health Cambridge   
10792                              CIC Health Providence   
10793                              Camden County College   
10794              Mohegan Sun - Thames Garage Bus Lobby   
10795                   Springfield Clinic Drive Up Labs   
10796             Mastery Charter School - Wister Campus   
10797                                Leman Middle School   
10798                                Leman Middle School   
10799                            Kahoka Christian Church   
10800                  Marshall County Health Department   
10801                                Premier Urgent Care   
10802                                    The Word Church   
10803                             Cherryland Center Mall   

                                            Full Address  \
0      Wheeling Park (Near Ice Rink), 1801 National R...   
1              147 South Main Street Middleton, MA 01949   
2      127 Anderson Street, Suite 101 Timber Court Bu...   
3      Palisades Int Interstate Parkway, Exit 17, Bea...   
4                  

In [7]:
_ = testing_sites_df['county_states'].unique
_

<bound method Series.unique of 0               Ohio , West Virginia
1              Essex , Massachusetts
2           Allegheny , Pennsylvania
3                  Orange , New York
4           Allegheny , Pennsylvania
5          Montgomery , Pennsylvania
6          Worcester , Massachusetts
7                  Ingham , Michigan
8                  Ingham , Michigan
9              Lehigh , Pennsylvania
10             Monmouth , New Jersey
11             Monroe , Pennsylvania
12             Carbon , Pennsylvania
13             Lehigh , Pennsylvania
14              Bucks , Pennsylvania
15               Herkimer , New York
16            Luzerne , Pennsylvania
17         Schuylkill , Pennsylvania
18                      Lucas , Ohio
19        Northampton , Pennsylvania
20             Lehigh , Pennsylvania
21        Northampton , Pennsylvania
22              St. Joseph , Indiana
23             Monroe , Pennsylvania
24              St. Joseph , Indiana
25              Chittenden , Vermont
26     

# Covid

In [8]:
Covid_data = pd.read_csv("https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-counties.csv")

In [9]:
Covid_data['mortality'] = Covid_data['deaths']/Covid_data['cases']
Covid_data.tail()

date      county    state     fips  cases  deaths  mortality
978926  2021-01-29  Sweetwater  Wyoming  56037.0   3464    32.0   0.009238
978927  2021-01-29       Teton  Wyoming  56039.0   3078     6.0   0.001949
978928  2021-01-29       Uinta  Wyoming  56041.0   1959    12.0   0.006126
978929  2021-01-29    Washakie  Wyoming  56043.0    863    25.0   0.028969
978930  2021-01-29      Weston  Wyoming  56045.0    610     4.0   0.006557

# Merge covid testing sites

In [10]:
testing_sites_covid = Covid_data.merge(testing_sites_df,
    how='inner',
    on = 'county'
)

testing_sites_covid

date     county       state     fips  cases  deaths  \
0         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
1         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
2         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
3         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
4         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
5         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
6         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
7         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
8         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
9         2020-01-21  Snohomish  Washington  53061.0      1     0.0   
10        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
11        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
12        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
13        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
14        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
15        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
16        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
17        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
18        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
19        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
20        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
21        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
22        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
23        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
24        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
25        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
26        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
27        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
28        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
29        2020-01-21  Snohomish  Washington  53061.0      1     0.0   
...              ...        ...         ...      ...    ...     ...   
13014555  2021-01-15      Hayes    Nebraska  31085.0     56     0.0   
13014556  2021-01-15      Hayes    Nebraska  31085.0     56     0.0   
13014557  2021-01-16      Hayes    Nebraska  31085.0     56     0.0   
13014558  2021-01-16      Hayes    Nebraska  31085.0     56     0.0   
13014559  2021-01-17      Hayes    Nebraska  31085.0     56     0.0   
13014560  2021-01-17      Hayes    Nebraska  31085.0     56     0.0   
13014561  2021-01-18      Hayes    Nebraska  31085.0     56     0.0   
13014562  2021-01-18      Hayes    Nebraska  31085.0     56     0.0   
13014563  2021-01-19      Hayes    Nebraska  31085.0     56     0.0   
13014564  2021-01-19      Hayes    Nebraska  31085.0     56     0.0   
13014565  2021-01-20      Hayes    Nebraska  31085.0     57     0.0   
13014566  2021-01-20      Hayes    Nebraska  31085.0     57     0.0   
13014567  2021-01-21      Hayes    Nebraska  31085.0     58     0.0   
13014568  2021-01-21      Hayes    Nebraska  31085.0     58     0.0   
13014569  2021-01-22      Hayes    Nebraska  31085.0     58     0.0   
13014570  2021-01-22      Hayes    Nebraska  31085.0     58     0.0   
13014571  2021-01-23      Hayes    Nebraska  31085.0     58     0.0   
13014572  2021-01-23      Hayes    Nebraska  31085.0     58     0.0   
13014573  2021-01-24      Hayes    Nebraska  31085.0     58     0.0   
13014574  2021-01-24      Hayes    Nebraska  31085.0     58     0.0   
13014575  2021-01-25      Hayes    Nebraska  31085.0     58     0.0   
13014576  2021-01-25      Hayes    Nebraska  31085.0     58     0.0   
13014577  2021-01-26      Hayes    Nebraska  31085.0     58     0.0   
13014578  2021-01-26      Hayes    Nebraska  31085.0     58     0.0   
13014579  2021-01-27      Hayes    Nebra

In [11]:
#  mort_county.corr()[['asthma_value_times_count', 'deaths']]

# Asthma

In [12]:
asthma = pd.read_csv('500_Cities__Current_asthma_among_adults_aged___18_years.csv')

In [13]:
asthma['state'] = asthma['StateDesc']
asthma['asthma_value_times_count'] = (asthma['Data_Value'] * asthma['PopulationCount'])/100
asthma['asthma_value_times_count']

0         528.612
1        5524.416
2         478.986
3         359.048
4         244.664
5         602.217
6         533.088
7         129.114
8         208.284
9         129.642
10        199.088
11        430.248
12       6983.556
13        301.770
14        858.104
15        477.752
16        122.636
17        486.128
18        298.067
19        448.128
20        565.785
21        521.489
22        267.696
23        565.779
24        545.008
25        315.375
26        538.160
27        152.724
28        195.306
29        378.990
           ...   
28976     462.580
28977    6506.056
28978     655.820
28979     792.820
28980     687.810
28981     170.655
28982     399.000
28983     375.800
28984     363.900
28985     142.784
28986     311.328
28987         NaN
28988      94.554
28989     447.282
28990      98.560
28991      87.045
28992     223.479
28993    6576.774
28994     433.100
28995     575.920
28996     674.432
28997     378.945
28998     491.470
28999     313.720
29000     

In [14]:
# asthma.drop(columns = ['Category','UniqueID','Measure','Short_Question_Text','Data_Value_Footnote','CategoryID','MeasureId','CityFIPS','TractFIPS'], inplace = True)
asthma.columns

Index(['Year', 'StateAbbr', 'StateDesc', 'CityName', 'GeographicLevel',
       'DataSource', 'Category', 'UniqueID', 'Measure', 'Data_Value_Unit',
       'DataValueTypeID', 'Data_Value_Type', 'Data_Value',
       'Low_Confidence_Limit', 'High_Confidence_Limit',
       'Data_Value_Footnote_Symbol', 'Data_Value_Footnote', 'PopulationCount',
       'GeoLocation', 'CategoryID', 'MeasureId', 'CityFIPS', 'TractFIPS',
       'Short_Question_Text', 'state', 'asthma_value_times_count'],
      dtype='object')

In [15]:
'''
changed aggfunc to mean because were dealing with percentage value, 
sum misrepresented the data
'''

asthma1 = asthma.groupby(['state']).mean()
asthma_state_count = asthma1['asthma_value_times_count']
asthma1

Year  Data_Value  Low_Confidence_Limit  \
state                                                     
Alabama        2017.0   11.425490             10.760224   
Alaska         2017.0    9.040351              8.621053   
Arizona        2017.0   10.344531              9.794141   
Arkansas       2017.0    9.733594              9.216406   
California     2017.0    8.897829              8.427422   
Colorado       2017.0    9.454213              8.948455   
Connecticut    2017.0   10.980085             10.433051   
Delaware       2017.0   12.065385             11.392308   
District of C  2017.0   10.266111              9.639444   
Florida        2017.0    8.749375              8.273769   
Georgia        2017.0    9.877253              9.292918   
Hawaii         2017.0    8.848729              8.496186   
Idaho          2017.0    9.471233              8.954795   
Illinois       2017.0    9.436790              8.872067   
Indiana        2017.0   11.068843             10.452612   
Iowa           2017.0    9.690594              9.169802   
Kansas         2017.0    9.502903              9.017097   
Kentucky       2017.0   10.698120             10.112406   
Louisiana      2017.0    9.966142              9.334908   
Maine          2017.0   10.359091              9.813636   
Maryland       2017.0   11.478109             10.761692   
Massachusetts  2017.0   11.256863             10.662157   
Michigan       2017.0   12.239153             11.531481   
Minnesota      2017.0    8.173457              7.733642   
Mississippi    2017.0   11.103797             10.379747   
Missouri       2017.0    9.904740              9.343567   
Montana        2017.0    9.414894              8.887234   
Nebraska       2017.0    8.663761              8.178899   
Nevada         2017.0   10.050678              9.542818   
New Hampshire  2017.0   11.100000             10.482353   
New Jersey     2017.0    9.969255              9.408075   
New Mexico     2017.0    9.806542              9.310748   
New York       2017.0   10.044032              9.471895   
North Carolin  2017.0    9.820833              9.283460   
North Dakota   2017.0    8.533333              8.003704   
Ohio           2017.0   11.116329             10.480691   
Oklahoma       2017.0   10.470190              9.941015   
Oregon         2017.0   10.383523              9.838068   
Pennsylvania   2017.0   11.165581             10.529302   
Rhode Island   2017.0   12.115094             11.507547   
South Carolin  2017.0    9.938418              9.330508   
South Dakota   2017.0    8.538333              8.070000   
Tennessee      2017.0   10.802294             10.161185   
Texas          2017.0    8.508445              8.013846   
United States  2017.0    8.950000              8.800000   
Utah           2017.0    9.296813              8.775299   
Vermont        2017.0   11.153846             10.338462   
Virginia       2017.0   10.224376              9.683877   
Washington     2017.0   10.088136              9.603766   
West Virginia  2017.0   11.266667             10.700000   
Wisconsin      2017.0   10.694203             10.070773   
Wyoming        2017.0    9.644444              9.155556   

               High_Confidence_Limit  PopulationCount      CityFIPS  \
state                                                                 
Alabama                    12.210084     7.620821e+03  1.383106e+05   
Alaska                      9.522807     1.535926e+04  2.030000e+05   
Arizona                    10.979785     1.099551e+04  4.530350e+05   
Arkansas                   10.329687     1.114484e+04  5.382832e+05   
California                  9.427933     1.196123e+04  6.480528e+05   
Colorado                   10.052247     1.023229e+04  8.280779e+05   
Connecticut                11.602542     1.107831e+04  9.456635e+05   
Delaware                   12.830769     8.175115e+03  1.077580e+06   
District of C              11.047222     9.973309e+03  1.150000e+06   
Florida                     9.306025     1.109481

In [16]:
#putting years in 1 column before merge
#didn't use this df because we have another thats grouped by state
asthma_premerge = asthma[['StateAbbr','Year','Data_Value']]


# Census -people with insurance

In [17]:

census_insurance = pd.read_csv('https://www2.census.gov/programs-surveys/demo/datasets/income-poverty/time-series/data-extracts/2018/cps-asec-bridge-file/hhpub18.csv')
census_insurance.columns
#HCOV NOW_HCOV 
# HTML(filename='www.census.gov/data/tables/time-series/demo/health-insurance/historical-series/hib.html')

Index(['H_IDNUM', 'GEREG', 'GESTFIPS', 'GEDIV', 'HRHTYPE', 'HEFAMINC',
       'H_MONTH', 'H_YEAR', 'H_TENURE', 'H_HHNUM',
       ...
       'THCHCARE_VAL', 'THPROP_VAL', 'GTCBSA', 'GTCO', 'GTCBSAST', 'GTCBSASZ',
       'GTCSA', 'GTMETSTA', 'GTINDVPC', 'MMYY'],
      dtype='object', length=134)

# Merging covid and asthma data

In [18]:
# Covid_data & asthma
var_of_interest= ['confirmed_deaths','confirmed_cases', ]

In [19]:
covid_asthma = pd.merge(asthma1,Covid_data,
    how="left",                      
    left_on = ['state'],
    right_on = ['state']) 
# covid_asthma.drop('United States', axis = 0)
covasdrop= ['fips','Low_Confidence_Limit','High_Confidence_Limit','CityFIPS','TractFIPS','Year','asthma_value_times_count','PopulationCount']
covid_asthma = covid_asthma.drop(columns = covasdrop)
covid_asthma = covid_asthma.rename(columns={'Data_Value':'perc pop with asthma','mortality':'covid mortality'})

In [20]:
asthmamort = covid_asthma.groupby('county').mean()
asthmamort.to_csv('asthmamort.csv')

In [21]:
covid_asthma.corr()[['perc pop with asthma', 'deaths']]

perc pop with asthma    deaths
perc pop with asthma              1.000000  0.013310
cases                            -0.019661  0.736290
deaths                            0.013310  1.000000
covid mortality                   0.026637  0.012677

# AQI

In [22]:
# #not updated through 12/2020
# with open('waqi-covid19-airqualitydata-2020.csv') as AQI:
#     AQI_older = pd.read_csv(AQI)
# AQI_older.head()


In [23]:
# with open('waqi-covid19-airqualitydata-2020.csv') as AQI:
#     AQI_old = pd.read_csv(AQI)
# AQI_old.head()


In [104]:
#this takes us to a document where we need to delete the first five rows bc it's just information 
#updated currently woohoo!
AQI = pd.read_excel('AQI_US.xlsx')
AQI.head()
AQI = AQI.sort_values('Date', ascending = False)
AQIbig6list=['so2','pm10','pm25','co','no2','o3']
AQI_big6df= AQI[AQI['Specie'].isin(AQIbig6list)]


# aqigroup=['City','Specie']
# AQI_average_values = AQI_big6df.groupby(by = aqigroup).mean()
# AQI.to_csv('AQI.csv')

In [105]:
'''
Replacing NY buroughs with 'New York' for the merge with city df
'''

NYBur={
    'Manhattan':'New York', 'Brooklyn':'New York','Queens':'New York', 
    'Staten Island':'New York', 'The Bronx':'New York'
}
AQI_big6df['City'] = AQI_big6df['City'].replace(to_replace = NYBur)

/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


In [107]:
AQI_average_values = pd.pivot_table(AQI_big6df, index = ['City'], values = 'median', columns = 'Specie',aggfunc = 'mean')


In [127]:
aqi_city['state_name'].unique()

array(['New Mexico', 'Georgia', 'Texas', 'Illinois', 'Indiana', 'Michigan',
       'Missouri', 'Kansas', 'Louisiana', 'Nebraska', 'Minnesota',
       'Arkansas', 'Pennsylvania', 'Nevada', 'Maryland', 'Ohio', 'Idaho',
       'Massachusetts', 'Virginia', 'Kentucky', 'New York', 'Wisconsin',
       'Iowa', 'North Carolina', 'Tennessee', 'South Carolina',
       'Mississippi', 'California', 'Alabama', 'New Jersey',
       'South Dakota', 'Montana', 'North Dakota', 'Oregon', 'Colorado',
       'Connecticut', 'Hawaii', 'Alaska', 'Delaware', 'Wyoming', 'Florida',
       'Vermont', 'West Virginia', 'Oklahoma', 'Arizona', 'Maine',
       'Rhode Island', 'Utah', nan, 'Puerto Rico', 'Washington'], dtype=object)

In [131]:


AQI_average_values['state_name'] = ['New Mexico','Georgia','Texas',
'Maryland','Idaho','Massachusetts', 'North Carolina','Illinois','South Carolina','Ohio','Texas','Colorado','Michigan','Texas',
'Indiana','California','Connecticut','Hawaii','Texas','Indiana','Mississippi','Florida',
'Nevada','Arkansas','California','Wisconsin','Tennessee','Florida','Wisconsin','Tennessee',
'New York','California','Oklahoma','Nebraska','Pennsylvania','Arizona','Oregon',
'Rhode Island','North Carolina','Virginia','California','Minnesota','Oregon','Utah',
'Texas','California','California','California','Washington','Illinois','Florida','Arizona','Maryland']
AQI_average_values

Specie                  co        no2         o3       pm10       pm25  \
City                                                                     
Albuquerque       1.832391   6.692545  29.788175  18.699229  24.411311   
Atlanta           2.841645   6.257326  19.140206  15.107969  35.149100   
Austin                 NaN   3.329897  20.808808        NaN  34.973890   
Baltimore              NaN   8.650129  21.242416  11.718579  30.686375   
Boise             2.930848   8.005656  20.900771  19.177835  33.064433   
Boston            2.328021   5.281491  22.412596        NaN  27.079692   
Charlotte         0.890909        NaN  20.126221  11.860215  30.637532   
Chicago           2.591599   7.649731  23.569409  14.219512  32.673522   
Columbia               NaN        NaN        NaN        NaN  36.890173   
Columbus          2.441714   7.434961  27.050442        NaN  29.125964   
Dallas            2.468056   4.697945  21.306202  18.202899  32.240741   
Denver            2.416195   7.392545  12.710283  21.226221  31.449871   
Detroit           4.062207   9.356667  21.996176  14.347962  36.186528   
El Paso           2.639041   6.294859  26.656812  23.989717  28.526992   
Fort Worth             NaN   5.044828  21.713506        NaN  29.694517   
Fresno            3.753985   9.300257  21.656555  37.219638  56.134367   
Hartford          2.751157   4.207455  11.455013  12.699229  29.447301   
Honolulu         20.435754        NaN  17.217784  10.270619  18.817010   
Houston           2.553237   8.162329  18.827835        NaN  37.038660   
Indianapolis      3.526992   7.291774  21.222108  17.373711  35.269923   
Jackson           1.402062   3.277246  20.524422  18.557143  35.393316   
Jacksonville      4.355556   3.686889  10.748843  17.624679  34.000000   
Las Vegas         0.100000   7.297429  14.794602  22.912596  27.668380   
Little Rock            NaN        NaN  18.790181        NaN  32.904393   
Los Angeles       4.759126  19.061954  15.965810  29.564767  52.730077   
Madison                NaN        NaN  26.505882  13.807198  33.041131   
Memphis                NaN        NaN  21.046272        NaN  31.591260   
Miami             1.407833        NaN  10.964781  15.100796  27.606684   
Milwaukee         2.608740   7.849357  23.490488  13.881748  32.876607   
Nashville              NaN        NaN  21.112821        NaN  33.491003   
New York          2.747068  13.064523  20.916667        NaN  27.636504   
Oakland           4.563753   9.581491  17.123136        NaN  36.956298   
Oklahoma City     5.953866   6.604884  24.465810  18.994859  36.246787   
Omaha             2.109021        NaN  19.186340  17.768831  30.007712   
Philadelphia      3.791176   6.597172  22.305141        NaN  33.483290   
Phoenix           4.621594   9.324679  19.580463  22.863753  32.012853   
Portland          2.717995   7.220051   3.273779        NaN  27.257069   
Providence        3.463896   9.263377  25.783465        NaN  25.992208   
Raleigh           2.661954   5.488946  20.378406  10.892216  27.463918   
Richmond          0.100000   2.043814  10.728354        NaN  28.976351   
Sacramento             NaN   4.079328  10.958355  34.989583  38.683117   
Saint Paul        2.714910   7.063753  18.948454  14.388175  29.616967   
Salem                  NaN        NaN  14.459043        NaN  30.059126   
Salt Lake City    1.969947  11.358355  24.119280        NaN  30.794344   
San Antonio            NaN   4.716438  21.862371        NaN  33.424870   
San Diego         4.285165  10.064481  18.688983        NaN  50.141631   
San Francisco     4.045758   6.866838  19.140360        NaN  34.390746   
San Jose          4.976607   9.523907  18.885347  15.140271  36.275064   
Seattle           2.998446  10.825450  17.599483        NaN  27.976864   
Springfield            NaN        NaN  22.374104        NaN  32.845758   
Tallahassee       1.488686        NaN  10.454005        NaN  35.994859   
Tucson            2.010825   6.232558  24.482262  22.601542  25.997429   
Wash

# Merging AQI with  city

Problem to overcome in the merge is that the AQI data geolocates through country and city while the covid_asthma dataset geolocates through state and county. How can we make a new column with only county or city or state for the dataset that lacks it?

We want to create a function that will sort the cities into the state

In [77]:
us_cities = pd.read_csv('uscities.csv')
us_cities = us_cities.rename(columns={'city':'City'})

In [100]:
aqi_city = AQI_average_values.merge(us_cities,
    how="left",                      
    left_on = ['City'],
    right_on = ['City']) 
aqi_city

array(['New Mexico', 'Georgia', 'Texas', 'Illinois', 'Indiana', 'Michigan',
       'Missouri', 'Kansas', 'Louisiana', 'Nebraska', 'Minnesota',
       'Arkansas', 'Pennsylvania', 'Nevada', 'Maryland', 'Ohio', 'Idaho',
       'Massachusetts', 'Virginia', 'Kentucky', 'New York', 'Wisconsin',
       'Iowa', 'North Carolina', 'Tennessee', 'South Carolina',
       'Mississippi', 'California', 'Alabama', 'New Jersey',
       'South Dakota', 'Montana', 'North Dakota', 'Oregon', 'Colorado',
       'Connecticut', 'Hawaii', 'Alaska', 'Delaware', 'Wyoming', 'Florida',
       'Vermont', 'West Virginia', 'Oklahoma', 'Arizona', 'Maine',
       'Rhode Island', 'Utah', nan, 'Puerto Rico', 'Washington'], dtype=object)

In [82]:
'''
dropping columns and renaming for clarity before merge
'''
aqicitydrop = ['source', 'military', 'incorporated', 'timezone','lat', 'lng','county_fips', 'Country', 'City', 'ranking', 'zips','id','state_id', 'state_name','city_ascii']
aqi_city = aqi_city.drop(columns= aqicitydrop)
aqi_city = aqi_city.rename(columns={'state_name':'state','county_name':'county'})


KeyError: "['source' 'military' 'incorporated' 'timezone' 'lat' 'lng' 'county_fips'\n 'Country' 'City' 'ranking' 'zips' 'id' 'state_id' 'state_name'\n 'city_ascii'] not found in axis"

In [78]:
us_cities

City           city_ascii state_id  \
0                 New York             New York       NY   
1              Los Angeles          Los Angeles       CA   
2                  Chicago              Chicago       IL   
3                    Miami                Miami       FL   
4                   Dallas               Dallas       TX   
5             Philadelphia         Philadelphia       PA   
6                  Houston              Houston       TX   
7                  Atlanta              Atlanta       GA   
8               Washington           Washington       DC   
9                   Boston               Boston       MA   
10                 Phoenix              Phoenix       AZ   
11                 Seattle              Seattle       WA   
12           San Francisco        San Francisco       CA   
13                 Detroit              Detroit       MI   
14               San Diego            San Diego       CA   
15             Minneapolis          Minneapolis       MN   
16                   Tampa                Tampa       FL   
17                  Denver               Denver       CO   
18                Brooklyn             Brooklyn       NY   
19                  Queens               Queens       NY   
20               Riverside            Riverside       CA   
21               Baltimore            Baltimore       MD   
22               Las Vegas            Las Vegas       NV   
23                Portland             Portland       OR   
24             San Antonio          San Antonio       TX   
25               St. Louis            St. Louis       MO   
26              Sacramento           Sacramento       CA   
27                 Orlando              Orlando       FL   
28                San Jose             San Jose       CA   
29               Cleveland            Cleveland       OH   
...                    ...                  ...      ...   
28342               McLeod               McLeod       ND   
28343                Anoka                Anoka       NE   
28344  Rock Island Arsenal  Rock Island Arsenal       IL   
28345                 Cave                 Cave       MO   
28346               Verdon               Verdon       SD   
28347         Prairie City         Prairie City       SD   
28348             Thurmond             Thurmond       WV   
28349         Lost Springs         Lost Springs       WY   
28350          Butterfield          Butterfield       TX   
28351            Livengood            Livengood       AK   
28352           Healy Lake           Healy Lake       AK   
28353               Beluga               Beluga       AK   
28354       Four Mile Road       Four Mile Road       AK   
28355               Alatna               Alatna       AK   
28356                 Ruso                 Ruso       ND   
28357             Brownlee             Brownlee       NE   
28358             Hoot Owl             Hoot Owl       OK   
28359             Freeport             Freeport       KS   
28360          St. Charles          St. Charles       SD   
28361                 Lily                 Lily       SD   
28362           Warm River           Warm River       ID   
28363           White Rock           White Rock       SD   
28364            Hillsview            Hillsview       SD   
28365          Birch Creek          Birch Creek       AK   
28366                Gross                Gross       NE   
28367            Poso Park            Poso Park       CA   
28368               Lotsee               Lotsee       OK   
28369            The Ranch            The Ranch       MN   
28370              Roswell              Roswell       SD   
28371               Monowi               Monowi       NE   

                 state_name  county_fips           county_name      lat  \
0                  New York        36061              New York  40.6943   
1                California         6037           Los Angeles  34.1139   
2                  Illinois        17031                  Cook  41.8373   
3

# Merging Covid/Asthma/AQI

1. aqi cityies have incorrect counties

In [107]:
aqi_city

Date Specie  count  min   max  median  variance        city  \
0      2021-01-21   pm25     43  1.0  59.0    21.0   1314.67   Manhattan   
1      2021-01-21   pm25     43  1.0  59.0    21.0   1314.67   Manhattan   
2      2021-01-21   pm25     43  1.0  59.0    21.0   1314.67   Manhattan   
3      2021-01-21   pm25     43  1.0  59.0    21.0   1314.67   Manhattan   
4      2021-01-21   pm10     54  1.0  32.0     6.0    462.54       Miami   
...           ...    ...    ...  ...   ...     ...       ...         ...   
467037 2019-12-30     co     23  1.4   1.9     1.7      0.15     Atlanta   
467038 2019-12-30     co     23  1.4   1.9     1.7      0.15     Atlanta   
467039 2019-12-30     co     23  1.4   1.9     1.7      0.15     Atlanta   
467040 2019-12-30     co     23  1.4   1.9     1.7      0.15     Atlanta   
467041 2019-12-30     o3     23  4.0  31.2    18.4    526.82  Fort Worth   

            county  population  density  
0         New York   1628706.0  27545.0  
1            Riley     57046.0   1062.0  
2             Will      8240.0    476.0  
3         Gallatin      1906.0    421.0  
4       Miami-Dade   6445545.0   5019.0  
...            ...         ...      ...  
467037       Macon       380.0    428.0  
467038      Cowley       181.0    146.0  
467039        Winn       146.0     52.0  
467040      Phelps       132.0    220.0  
467041     Tarrant    909585.0   1016.0  

[467042 rows x 11 columns]

In [106]:
covid_asthma

state  perc pop with asthma      CityFIPS     TractFIPS        date  \
0       Alabama             11.425490  1.383106e+05  1.092529e+09  2020-03-13   
1       Alabama             11.425490  1.383106e+05  1.092529e+09  2020-03-13   
2       Alabama             11.425490  1.383106e+05  1.092529e+09  2020-03-13   
3       Alabama             11.425490  1.383106e+05  1.092529e+09  2020-03-13   
4       Alabama             11.425490  1.383106e+05  1.092529e+09  2020-03-13   
...         ...                   ...           ...           ...         ...   
903908  Wyoming              9.644444  5.613900e+06  5.602100e+10  2021-01-27   
903909  Wyoming              9.644444  5.613900e+06  5.602100e+10  2021-01-27   
903910  Wyoming              9.644444  5.613900e+06  5.602100e+10  2021-01-27   
903911  Wyoming              9.644444  5.613900e+06  5.602100e+10  2021-01-27   
903912  Wyoming              9.644444  5.613900e+06  5.602100e+10  2021-01-27   

            county   cases  deaths  covid mortality  
0           Elmore     1.0     0.0         0.000000  
1        Jefferson     2.0     0.0         0.000000  
2        Limestone     1.0     0.0         0.000000  
3       Montgomery     1.0     0.0         0.000000  
4       Tuscaloosa     1.0     0.0         0.000000  
...            ...     ...     ...              ...  
903908  Sweetwater  3428.0    32.0         0.009335  
903909       Teton  3020.0     6.0         0.001987  
903910       Uinta  1948.0    12.0         0.006160  
903911    Washakie   863.0    25.0         0.028969  
903912      Weston   610.0     4.0         0.006557  

[903913 rows x 9 columns]

In [104]:
master1 = pd.merge(covid_asthma,aqi_city,
    how="left",                      
    left_on = ['county'],
    right_on = ['county']
)
master1

MemoryError: Unable to allocate 5.62 GiB for an array with shape (753807573,) and data type int64

Ideas on kernel death:
1. if joining by state, different cities will be joined together? --> refine to match by city or county instead of state probably by county

2.  5.62 GiB for array with shape 753807573


# HistAQI


In [ ]:
histAQI = pd.read_excel('airqualitytrendsbycity1990-2019.xlsx')
histAQI.head()

In [ ]:
histAQI_Year_column = pd.melt(histAQI, id_vars=(['Pollutant','StateAbbr','Trend Statistic']), value_vars=([1990,
                                1991,                          1992,
                                1993,                          1994,
                                1995,                          1996,
                                1997,                          1998,
                                1999,                          2000,
                                2001,                          2002,
                                2003,                          2004,
                                2005,                          2006,
                                2007,                          2008,
                                2009,                          2010,
                                2011,                          2012,
                                2013,                          2014,
                                2015,                          2016,
                                2017,                          2018,
                                2019]), var_name='Year', value_name='Pollutant Value').dropna()
histAQI_Year_column.head()

In [ ]:
cities = histAQI['Core Based Statistical Area']
y = [str(i).split() for i in cities]

states = []
for i in y:
    state = i[-1]
    states.append(state)
histAQI['StateAbbr'] = states

In [ ]:
histAQI.head()

In [ ]:
histAQI_asthma= histAQI_asthma[histAQI_asthma.StateAbbr != 'US']

In [ ]:
histAQI_asthma

In [ ]:
#copied code to fill in values from histAQI_asthma
sns.lineplot(x = "Year", y = "Pollutant Value", data = histAQI_Year_column)
plt.title('Total Pollutant Values over Time')
plt.show()

In [ ]:
sns.lineplot(x = "Year", y = "Pollutant Value", data = histAQI_Year_column)
plt.title('Total Pollutant Values over Time')
plt.show()

In [ ]:
sns.lineplot(x = "Year", y = "Pollutant Value", data = histAQI_Year_column, hue = "Pollutant")
plt.title('Pollutant Values over Time')
plt.show()

# Merge AQI and Asthma

In [ ]:
#append adds rows
# histAQI_asthma = histAQI_Year_column.append(asthma_premerge)

#merge combines
AQI_asthma = pd.merge(asthma1,AQI_big6df,
    how="left",                      
    left_on = ['state'],
    right_on = ['state'])                        
AQI_asthma


# Demographics


In [ ]:
demographics_data = pd.read_csv("https://data.cdc.gov/resource/vbim-akqf.csv")

In [ ]:
demographics_data.groupby('race_ethnicity_combined').size()


In [ ]:
demographics_data.describe()

Date Country             City       Specie  count     min     max  \
612567  9/9/2020      US         Hartford         pm10     48    10.0    32.0   
655525  9/9/2020      US      Springfield           o3     24     1.6    32.8   
622276  9/9/2020      US          Houston          no2     93     0.3    23.3   
587151  9/9/2020      US         Portland     pressure    188    29.5  1017.9   
636160  9/9/2020      US        Charlotte         pm25     47     5.0    59.0   
611427  9/9/2020      US         Hartford     humidity     48    48.0    91.7   
703393  9/9/2020      US            Miami    wind-gust     55     0.2    11.0   
576690  9/9/2020      US           Boston         pm25    133     5.0    65.0   
622961  9/9/2020      US          Houston     pressure    114  1010.2  1018.0   
607360  9/9/2020      US        Milwaukee         pm10    119     1.0     8.0   
606610  9/9/2020      US        Milwaukee          dew     28    10.0    13.0   
573099  9/9/2020      US          Jackson          no2     22     1.0     9.1   
765115  9/9/2020      US           Queens           o3     69     0.8    35.2   
617639  9/9/2020      US          Chicago   wind-speed    206     0.1    10.3   
723260  9/9/2020      US        San Diego   wind-speed     15     0.2     6.5   
621816  9/9/2020      US          Houston           o3    118     0.8    38.4   
641631  9/9/2020      US          Oakland          no2    122     1.9    19.5   
715396  9/9/2020      US        Las Vegas    wind-gust    148     0.1    20.1   
718712  9/9/2020      US      San Antonio  temperature    110    17.3    30.5   
639749  9/9/2020      US       Sacramento     humidity     62    23.6    74.5   
606558  9/9/2020      US        Milwaukee     pressure    167  1019.8  1032.1   
727389  9/9/2020      US         Columbus          dew     23    17.0    20.0   
675778  9/9/2020      US      Little Rock   wind-speed     46     1.5     5.1   
619253  9/9/2020      US          Chicago  temperature    215     8.6    19.0   
708818  9/9/2020      US     Jacksonville   wind-speed     67     0.1     4.1   
735394  9/9/2020      US           Denver  temperature    209    -0.6     8.0   
766005  9/9/2020      US           Queens     pressure     96  1022.9  1026.7   
745504  9/9/2020      US      Albuquerque     pressure     72  1010.0  1027.1   
632196  9/9/2020      US          Atlanta   wind-speed     15     0.1     1.5   
592796  9/9/2020      US           Austin     humidity     54    86.7    97.0   
...          ...     ...              ...          ...    ...     ...     ...   
610186  1/1/2020      US         Hartford         pm25     47     9.0    32.0   
634897  1/1/2020      US        Charlotte           o3     46     2.4    32.8   
762507  1/1/2020      US           Tucson          no2     24     2.9    19.3   
677220  1/1/2020      US        Baltimore          so2      2     1.5     1.5   
561504  1/1/2020      US    Oklahoma City         pm10     96     4.0    18.0   
616371  1/1/2020      US  Washington D.C.          so2     23     0.2     0.6   
656861  1/1/2020      US      Springfield   wind speed      8     2.1     4.0   
745679  1/1/2020      US    San Francisco          no2     68     1.0    22.3   
652201  1/1/2020      US         Brooklyn         pm25    168     1.0    54.0   
568132  1/1/2020      US          Raleigh         pm10     24     2.0    21.0   
626120  1/1/2020      US     Indianapolis          no2     22     5.6    10.2   
721224  1/1/2020      US     Philadelphia    wind-gust     22     1.6    10.8   
593219  1/1/2020      US           Austin         pm25     48     8.0   127.0   
720255  1/1/2020      US     Philadelphia           co     20     1.2    10.0   
725437  1/1/2020      US         Columbus         pm25     24     4.0    35.0   
688433  1/1/2020      US          El Paso           o3    166     0.8    35.2   
680905  1/1/2020      US          Phoenix         pm10    190     1.0   473.0   
641618  1/1/2020      US          Oakland  